pkg/api/v1/client/proxy.go

```diff
@@ -392,3 +397,52 @@ func (c ProxyScriptsAPI) makeDeleteRequest(uri string, isContentExpected bool) e
 
 	return nil
 }
+
+func (c ProxyScriptsAPI) GetExecutionArtifacts(executionID string) (artifacts testkube.Artifacts, err error) {
+	uri := c.getURI("/executions/%s/artifacts", executionID)
+	req := c.GetProxy("GET").
+		Suffix(uri)
+	resp := req.Do(context.Background())
+
+	if err := c.responseError(resp); err != nil {
+		return artifacts, fmt.Errorf("api/list-artifacts returned error: %w", err)
+	}
+
+	return c.getArtifactsFromResponse(resp)
+
+}
+func (c ProxyScriptsAPI) DownloadFile(executionID, fileName, destination string) (artifact string, err error) {
+	uri := c.getURI("/executions/%s/artifacts/%s", executionID, fileName)
+	req := c.GetProxy("GET").
+		Suffix(uri)
+
+	resp := req.Do(context.Background())
```

Comment:
Use Stream() instead. It'll return io.ReadCloser look at Logs() method for details

Reply:
the API endpoint is sending file as c.SendFile since c.SendStream was not working https://github.com/kubeshop/testkube/blob/cli_artifacts/internal/app/api/v1/scripts.go#L484

Reply:
fixed it